### PR TITLE
Add tracking of user responses for Analytics

### PR DESCRIPTION
**Pull request by RamAnanth**
Created a preliminary method in controller.py to save the user responses into categories (The results can be found in responses.csv). 
This can later be processed for analytics such as tracking how many times the bot got the correct action

### CODE REVIEW

File: weblm/controller.py [hunk 1]
```diff
@@ -196,6 +197,7 @@ def __init__(self, co: cohere.Client, objective: str):
         self.objective = objective
         self.previous_commands: List[str] = []
         self.moments: List[Tuple[str, str]] = []
+        self.user_responses:DefaultDict[str, int] = defaultdict(int)
```
Comment (aidangomez) [inline weblm/controller.py]:
```suggestion
        self.user_responses: DefaultDict[str, int] = defaultdict(int)
```

File: weblm/controller.py [hunk 2]
```diff
@@ -212,7 +214,7 @@ def reset_state(self):
     def success(self):
         for state, command in self.moments:
             self._save_example(state, command)
-
+
```
Comment (aidangomez) [inline weblm/controller.py]: can you use `yapf --style "{based_on_style: google, indent_width: 4, column_limit: 120}" -ir .` to format this repo?

File: weblm/main.py [hunk 3]
```diff
@@ -13,7 +13,8 @@
 import cohere
 
 from .controller import Command, Controller, Prompt
-from .crawler import URL_PATTERN, Crawler
+from .crawler import URL_PATTERN, Crawler, AsyncCrawler
```
Comment (aidangomez) [inline weblm/main.py]: Is AsyncCrawler used?

File: weblm/controller.py (hunk 2 above)
Comment (RamAnanth) [reply weblm/controller.py]: Sure, Sorry about the lack of proper formatting before. I formatted and pushed the changes

**Comment by aidangomez** [comment]
Looks like there are some conflicts to resolve but otherwise lgtm

**Review by aidangomez — APPROVED** [review]
LGTM

**Pull request merged**